modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/LeftSidebar/LeftSidebarTreeView.tsx

```diff
@@ -0,0 +1,394 @@
+		objectDefinitionId,
+		objectFolderName,
+	}: {
+		objectDefinitionId: number;
+		objectFolderName: string;
+	}) => {
+		const objectFolders = await API.getAllObjectFolders();
+
+		const currentObjectFolder = objectFolders.find(
+			(objectFolder) => objectFolder.name === objectFolderName
+		);
+
+		const objectDefinitionsFilteredByObjectFolder = await API.getObjectDefinitions(
+			`filter=objectFolderExternalReferenceCode eq '${currentObjectFolder?.externalReferenceCode}'`
+		);
+
+		const objectDefinition = objectDefinitionsFilteredByObjectFolder.find(
+			(objectDefinition) => objectDefinition.id === objectDefinitionId
+		) as ObjectDefinitionNodeData;
+
+		if (objectDefinition) {
+			const movedObjectDefinition: ObjectDefinitionNodeData = {
+				...objectDefinition,
+				objectFolderExternalReferenceCode:
+					selectedObjectFolder.externalReferenceCode,
+			};
+
+			try {
+				const newObjectDefinition = (await API.save({
+					item: movedObjectDefinition,
+					method: 'PATCH',
+					returnValue: true,
+					url: `/o/object-admin/v1.0/object-definitions/${objectDefinition?.id}`,
+				})) as ObjectDefinition;
+
+				dispatch({
+					payload: {
+						newObjectDefinition,
+						selectedObjectFolderName: selectedObjectFolder.name,
+					},
+					type: TYPES.ADD_OBJECT_DEFINITION_TO_OBJECT_FOLDER,
+				});
+
+				if (!objectDefinition.linkedObjectDefinition) {
+					dispatch({
+						payload: {
+							currentObjectFolderName: currentObjectFolder!.name,
+							deletedObjectDefinitionName:
+								newObjectDefinition.name,
+						},
+						type: TYPES.DELETE_OBJECT_DEFINITION,
+					});
+				}
+
+				openToast({
+					message: sub(
+						Liferay.Language.get('x-was-moved-successfully'),
+						`<strong>${getLocalizableLabel(
+							objectDefinition.defaultLanguageId,
+							movedObjectDefinition.label
+						)}</strong>`
+					),
+					type: 'success',
+				});
+			}
+			catch (error) {}
+		}
+	};
+
+	const onClickGoToFolder = (selectedObjectFolderName: string) => {
+		dispatch({
+			payload: {
+				objectFolderName: selectedObjectFolderName,
+			},
+			type: TYPES.SET_OBJECT_FOLDER_NAME,
+		});
+	};
+
+	const leftSidebarOtherObjectFoldersItems = filteredLeftSidebarItems.filter(
+		(filteredLeftSidebarItem) =>
+			filteredLeftSidebarItem.objectFolderName !==
+			selectedObjectFolder.name
+	);
+
+	const leftSidebarSelectedObjectFolderItem = filteredLeftSidebarItems.find(
+		(filteredLeftSidebarItem) =>
+			filteredLeftSidebarItem.objectFolderName ===
+			selectedObjectFolder.name
+	) as LeftSidebarItem;
+
+	const linkedObjectDefinitions = leftSidebarSelectedObjectFolderItem.leftSidebarObjectDefinitionItems?.filter(
+		(leftSidebarObjectDefinitionItem) =>
+			leftSidebarObjectDefinitionItem.type === 'linkedObjectDefinition'
+	);
+
+	const newOtherObjectFolders = leftSidebarOtherObjectFoldersItems.map(
+		(leftSidebarObjectFolderItem) => {
+			const objectDefinitions = leftSidebarObjectFolderItem.leftSidebarObjectDefinitionItems?.map(
```

Comment:
This map will not return a array of objectDefinition. So I think we can use a better name here.